File: maddie-carlson/day-11/shortest-word.js
function findShort(s) {
  //function takes a string "s" and returns the length (a number) of the shortest word in that string
  //"string will never be empty and you do not need to account for different data types", apparently

  //start by splitting the string via its spaces into an array to identify words
  let array = s.split(" ")
  let shortest = 1000

  //then iterate over that array in a loop, checking the length of each word and storing the smallest value found
  for (let i = 0; i < array.length; i++) {
    if (array[i].length < shortest) {
      shortest = array[i].length
    }
  }

  //return smallest value
  return shortest
}
/*
  Ember's Feedback:
  - Good job!
  - Good pseudocode and solution
*/
// Alternative Solution
function findShort(s) {
  return Math.min(...s.split(" ").map(word => word.length));
}

const findShort = (s) => s
  .split(' ')
  .sort((a, b) => b.length - a.length)
  .pop()
  .length;

File: caitlin-schaeffer/day-28/shortest-word.py
# Instructions: Simple, given a string of words, return the length of the shortest word(s). String will never be empty and you do not need to account for different data types.
# Examples: "bitcoin take over the world maybe who knows perhaps") = 3
# Params: given a string
# Return: integer
# Concerns: none
# Solution: 

# Given Code: 
# def find_short(s):
#     # your code here
#     return l # l: shortest word length

def find_short(s):
    words = s.split()
    sort = sorted(words, key=len)
    firstWord = sort[0]
    print("firstWord:", firstWord)
    l = len(firstWord)
    print("l:", l)
    return l
